CMakeLists.txt

```diff
@@ -139,18 +107,22 @@ ENDIF()
 #
 #  Cmake configuration
 #
-MESSAGE("Setting up the CMake configuration...")
+MESSAGE(STATUS "Setting up the CMake configuration...")
 IF(NOT CMAKE_BUILD_TYPE)
   SET(CMAKE_BUILD_TYPE "RELEASE")
-  SET(CMAKE_CXX_FLAGS "-fPIC -O2 -fpermissive -w -I ${Boost_INCLUDE_DIRS} -fopenmp")

```

Comment:
why to remove these flags?

Reply:
oh, i see you set them outside the if block; but again - the fpermissive was important at some point

Reply:
-fpermissive is not a safe option. Better to avoid it. Looks like it was a main issue with AppleClang compiler